### PR TITLE
Run tests with prover enabled. 

### PR DESCRIPTION
**Pull request by bkolad**
# Description
This PR introduces the following changes:
1. Moves `AppVerifier` to `demo-stf` which breaks the `demo-prover -> demo-rollup` dependency.
2. Adds `methods/guest-mock` a prover that works with `MockDa`
3. Makes uses of the `guest-mock` prover in `demo-rollup` tests with the `DemoProverConfig::Execute` setting. 
4. Updates CI accordingly.
5. Removes `ide_steup` readme, is it still relevant?
6. Removes the `local` feature as it is relevant only for the `EthSequencer`
## Testing
Ci passes.

## Docs
No changes.

### CODE REVIEW

File: examples/demo-prover/methods/guest-mock/riscv32im-risc0-zkvm-elf.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (preston-evans98) [inline examples/demo-prover/methods/guest-mock/riscv32im-risc0-zkvm-elf.json]: I think we can delete this file too. That's also a holdover from the old Risc0 version

**Review by preston-evans98 — COMMENTED** [review]
Looks good to me, except that it feels like we might be breaking an abstraction by adding a dependency on `sov-stf-runner` in `demo-stf`. Maybe we need to revisit something there?

**Comment by bkolad** [comment]
> Looks good to me, except that it feels like we might be breaking an abstraction by adding a dependency on `sov-stf-runner` in `demo-stf`. Maybe we need to revisit something there?

The underlying issue is that at some point we have to connect `sov_stf_runner::StateTransitionVerifier` with `demo_stf::Runtime`. The above approach adds the `sov-stf-runner -> demo-stf`  and assembles the types in the `demo-stf`. Alternativly we can import `sov_stf_runner & demo_stf` in both `demo-rollup` & `demo-prover` and create a 
```
pub type AppVerifier<DA, Zk> = StateTransitionVerifier<
    AppTemplate<
        ZkDefaultContext,
        <DA as DaVerifier>::Spec,
        Zk,
        Runtime<ZkDefaultContext, <DA as DaVerifier>::Spec>,
    >,
    DA,
    Zk,
>;
```
type in each crate but the downside is that we  will repeat the same type alias 2x.

**Review by preston-evans98 — APPROVED** [review]
Synced with @bkolad. We're going to merge this as-is and fix the abstraction in a follow-up

File: examples/demo-prover/methods/guest-mock/riscv32im-risc0-zkvm-elf.json (hunk 1 above)
Comment (bkolad) [reply examples/demo-prover/methods/guest-mock/riscv32im-risc0-zkvm-elf.json]: fixed

**Pull request merged**